jupyter_sigplot/sigplot.py

```diff
@@ -217,6 +188,42 @@ def overlay_file(self, path):
         self.overlay_href(path)
 
 
+
+def _prepare_array_input(input, layer_type, subsize):
```

Comment:
Thanks for popping this out into another function, much more modular!

You'll probably want to write a test or two against this function as well as fix the tests that broke due to this function.